File: Practice/268-missing-number/missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        sum = 0
        for i in nums:
            sum = sum + i
        
        n = len(nums)
        Sn = int((n*(n+1))/2)

        ans = Sn - sum

        return ans

        

File: Practice/268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        //striver a2z array-easy

        int n = nums.size();    //same for both approach
        // int sum=0;

        // for(int i=0; i<n; i++){
        //     sum += nums[i];
        // }

        // long Sn = (n*(n+1))/2;  //sum of n numbers

        // return Sn-sum;
        int xor1=0, xor2=0; //a^a=0  a^0=a

        for(int i=0; i<n; i++){ //better as we dont take long
            xor1 = xor1^(i+1);
            xor2 = xor2^nums[i];
        }

        return xor1^xor2;

    }
};
